# After applying this patch, enable system SQLite from the CCDB source root with:
#   cmake -S . -B build -DCCDB_USE_SYSTEM_SQLITE=ON
#   cmake --build build
#
# ON requires the SQLite development headers and library, located by CMake's FindSQLite3 module.
# For a nonstandard SQLite installation, also pass -DCMAKE_PREFIX_PATH=/absolute/path/to/sqlite.
# The flag defaults to OFF, which keeps CCDB's bundled SQLite amalgamation.
# GEMC's Meson builds enable the flag automatically for bundled CCDB.
#
diff --git a/src/Benchmarks/CMakeLists.txt b/src/Benchmarks/CMakeLists.txt
index f67a2c6..04f2e44 100644
--- a/src/Benchmarks/CMakeLists.txt
+++ b/src/Benchmarks/CMakeLists.txt
@@ -4,7 +4,9 @@ project(CCDB_bn_sqlite)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 
 include_directories("../../include")
-include_directories("../../include/SQLite")
+if(NOT CCDB_USE_SYSTEM_SQLITE)
+    include_directories("../../include/SQLite")
+endif()
 
 find_package (Threads)
 
@@ -14,4 +16,4 @@ set(SOURCE_FILES
 
 
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
-target_link_libraries(${PROJECT_NAME} ${CMAKE_THREAD_LIBS_INIT} ccdb ccdb_sqlite)
+target_link_libraries(${PROJECT_NAME} ${CMAKE_THREAD_LIBS_INIT} ccdb ${CCDB_SQLITE_TARGET})
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 94d050a..d135e11 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,11 @@
 cmake_minimum_required(VERSION 3.5)
 project(CCDB)
 
+if(POLICY CMP0200)
+    set(CMAKE_POLICY_DEFAULT_CMP0200 NEW)
+    cmake_policy(SET CMP0200 NEW)
+endif()
+
 set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
 set(CMAKE_DISABLE_SOURCE_CHANGES  ON)
 
@@ -57,8 +57,22 @@ endif()
 include(${CMAKE_CURRENT_LIST_DIR}/FindMySQL.cmake)
 
 add_definitions(-DCCDB_MYSQL)
+option(CCDB_USE_SYSTEM_SQLITE "Use the system SQLite library instead of the bundled amalgamation" OFF)
+
+if(CCDB_USE_SYSTEM_SQLITE)
+    find_package(SQLite3 REQUIRED)
+    if(TARGET SQLite3::SQLite3)
+        set(CCDB_SQLITE_TARGET SQLite3::SQLite3)
+    else()
+        set(CCDB_SQLITE_TARGET SQLite::SQLite3)
+    endif()
+    set(CCDB_SQLITE_STATIC_TARGET ${CCDB_SQLITE_TARGET})
+else()
+    add_subdirectory(SQLite)
+    set(CCDB_SQLITE_TARGET ccdb_sqlite)
+    set(CCDB_SQLITE_STATIC_TARGET ccdb_sqlite_static)
+endif()
 
-add_subdirectory(SQLite)
 add_subdirectory(Library)
 add_subdirectory(Tests)
 add_subdirectory(Benchmarks)
diff --git a/src/Library/CMakeLists.txt b/src/Library/CMakeLists.txt
index 4b496ef..ef9a197 100644
--- a/src/Library/CMakeLists.txt
+++ b/src/Library/CMakeLists.txt
@@ -4,7 +4,9 @@ project(ccdb VERSION 1.7)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 
 include_directories("../../include")
-include_directories("../../include/SQLite")
+if(NOT CCDB_USE_SYSTEM_SQLITE)
+    include_directories("../../include/SQLite")
+endif()
 include_directories(${MYSQL_INCLUDE_DIR})
 
 set(SOURCE_FILES
@@ -63,4 +65,4 @@
 
-target_link_libraries(ccdb ${MYSQL_LIBRARIES} ccdb_sqlite)
-target_link_libraries(ccdb_static ${MYSQL_LIBRARIES} ccdb_sqlite_static)
+target_link_libraries(ccdb ${MYSQL_LIBRARIES} ${CCDB_SQLITE_TARGET})
+target_link_libraries(ccdb_static ${MYSQL_LIBRARIES} ${CCDB_SQLITE_STATIC_TARGET})
 
diff --git a/src/Library/ccdbConfig.cmake.in b/src/Library/ccdbConfig.cmake.in
index 09b18fb..18440c6 100644
--- a/src/Library/ccdbConfig.cmake.in
+++ b/src/Library/ccdbConfig.cmake.in
@@ -12,4 +12,7 @@
 include(CMakeFindDependencyMacro)
 find_dependency(MySQL REQUIRED)
 find_dependency(Threads REQUIRED)
+if(@CCDB_USE_SYSTEM_SQLITE@)
+    find_dependency(SQLite3 REQUIRED)
+endif()
 include("${CMAKE_CURRENT_LIST_DIR}/ccdbTargets.cmake")
diff --git a/src/Tests/CMakeLists.txt b/src/Tests/CMakeLists.txt
index e560601..1bdd9c8 100644
--- a/src/Tests/CMakeLists.txt
+++ b/src/Tests/CMakeLists.txt
@@ -4,7 +4,9 @@ project(ccdb_tests)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
 include_directories("../../include")
-include_directories("../../include/SQLite")
+if(NOT CCDB_USE_SYSTEM_SQLITE)
+    include_directories("../../include/SQLite")
+endif()
 include_directories(${MYSQL_INCLUDE_DIR})
 
 find_package (Threads)
@@ -37,7 +39,6 @@ set(SOURCE_FILES
 
 
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
-target_link_libraries(${PROJECT_NAME} ${CMAKE_THREAD_LIBS_INIT} ccdb ccdb_sqlite)
+target_link_libraries(${PROJECT_NAME} ${CMAKE_THREAD_LIBS_INIT} ccdb ${CCDB_SQLITE_TARGET})
 add_dependencies(${PROJECT_NAME} ccdb)
 
-
